source/common/upstream/upstream_impl.cc

```diff
@@ -434,14 +414,52 @@ ClusterSharedPtr ClusterImplBase::create(const envoy::api::v2::Cluster& cluster,
   return std::move(new_cluster);
 }
 
+Stats::ScopePtr ClusterImplBase::generateStatsScope(const envoy::api::v2::Cluster& config,
```

Comment:
Not need to be a member function.  It can be a global function in the annoymous namespace.